src/test/java/io/jenkins/docker/connector/DockerComputerSSHConnectorTest.java

```diff
@@ -1,32 +1,67 @@
 package io.jenkins.docker.connector;
 
+import com.cloudbees.plugins.credentials.SystemCredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
 import com.github.dockerjava.api.command.CreateContainerCmd;
 import com.github.dockerjava.api.model.ExposedPort;
 import com.github.dockerjava.api.model.PortBinding;
 import com.github.dockerjava.api.model.Ports;
 import com.github.dockerjava.core.command.CreateContainerCmdImpl;
 import com.nirima.jenkins.plugins.docker.DockerTemplate;
 import com.nirima.jenkins.plugins.docker.DockerTemplateBase;
+import com.trilead.ssh2.signature.RSAKeyAlgorithm;
+
+import hudson.plugins.sshslaves.verifiers.NonVerifyingKeyVerificationStrategy;
 import io.jenkins.docker.client.DockerAPI;
+import jenkins.bouncycastle.api.PEMEncodable;
+
+import org.jenkinsci.main.modules.instance_identity.InstanceIdentity;
 import org.junit.Assert;
 import org.junit.Test;
 import org.mockito.Mockito;
 
+import static hudson.remoting.Base64.encode;
+
 import java.io.IOException;
 import java.util.Map;
-import java.util.concurrent.ExecutionException;
 
 public class DockerComputerSSHConnectorTest extends DockerComputerConnectorTest {
 
-    @Test
-    public void should_connect_agent() throws InterruptedException, ExecutionException, IOException {
+    private static final String SSH_SLAVE_IMAGE_IMAGENAME = "jenkins/ssh-slave";
+    private static final String SSH_SLAVE_IMAGE_JAVAPATH = "/usr/local/openjdk-8/bin/java";
 
+    @Test
+    public void connectAgentViaSSHUsingInjectSshKey() throws Exception {
+        final DockerComputerSSHConnector.SSHKeyStrategy sshKeyStrategy = new DockerComputerSSHConnector.InjectSSHKey(COMMON_IMAGE_USERNAME);
+        final DockerComputerSSHConnector connector = new DockerComputerSSHConnector(sshKeyStrategy);
+        connector.setJavaPath(SSH_SLAVE_IMAGE_JAVAPATH);
```

Comment:
The lack of this is what broke the test - the way this plugin invokes the container, the PATH doesn't have java on it within the SSH sessions (arguably a bug in the container image).